storage/remote/client.go

```diff
@@ -187,3 +190,70 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 
 	return resp.Results[0], nil
 }
+
+type labelsResponse struct {
+	Data     []string `json:"data,omitempty"`
+	Error    string   `json:"error,omitempty"`
+	Warnings []string `json:"warnings,omitempty"`
+}
+
+// makeLabelsRequest makes the actual request for /label and /labels/{name}/values from the remote client
+func (c *Client) makeLabelsRequest(ctx context.Context, url string) ([]string, storage.Warnings, error) {
```

Comment:
Wonder, can we use https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go#L235 instead?